static/js/people.js

```diff
@@ -17,24 +18,30 @@ exports.init = function () {
     // (all people we've seen), but people_dict can have duplicate
     // keys related to email changes.  We want to deprecate
     // people_dict over time and always do lookups by user_id.
-    people_dict = new Dict({fold_case: true});
-    people_by_name_dict = new Dict({fold_case: true});
-    people_by_user_id_dict = new Dict();
+    people_dict = new FoldDict();
+    people_by_name_dict = new FoldDict();
+    people_by_user_id_dict = new IntDict();
 
     // The next dictionary includes all active users (human/user)
     // in our realm, but it excludes non-active users and
     // cross-realm bots.
-    active_user_dict = new Dict();
-    cross_realm_dict = new Dict(); // keyed by user_id
-    pm_recipient_count_dict = new Dict();
+    active_user_dict = new IntDict();
+    cross_realm_dict = new IntDict(); // keyed by user_id
+    pm_recipient_count_dict = new IntDict();
 
-    // The next Dict maintains a set of ids of people with same full names.
-    duplicate_full_name_data = new Dict({fold_case: true});
+    // This maintains a set of ids of people with same full names.
+    duplicate_full_name_data = new FoldDict();
 };
 
 // WE INITIALIZE DATA STRUCTURES HERE!
 exports.init();
 
+function map_int(lst) {
+    return _.map(lst, function (s) {
+        return parseInt(s, 10);
+    });
+}
```

Comment:
When written as `lst.map(s => parseInt(s, 10))`, this doesn’t seem worth extracting as a helper function.

Reply:
keeping as is, `parseInt` has evil behavior if you call it wrong, I want to encapsulate it